crates/turborepo-lib/src/daemon/connector.rs

```diff
@@ -0,0 +1,548 @@
+            Ok(())
+        );
+    }
+
+    #[tokio::test]
+    async fn handles_kill_dead_server_missing_process() {
+        let tmp_dir = tempfile::tempdir().unwrap();
+        let tmp_path = tmp_dir.path().to_owned();
+
+        let pid = pid_path(&tmp_path);
+        std::fs::write(&pid, usize::MAX.to_string()).unwrap();
+        let sock = sock_path(&tmp_path);
+        std::fs::write(&sock, "").unwrap();
+
+        let connector = DaemonConnector {
+            pid_file: pid,
+            sock_file: sock,
+            can_kill_server: false,
+            can_start_server: false,
+        };
+
+        assert_matches!(
+            connector.kill_dead_server(Pid::from(usize::MAX)).await,
+            Ok(())
+        );
+
+        assert!(!connector.pid_file.exists(), "pid file should be deleted");
+    }
+
+    #[tokio::test]
+    async fn handles_kill_dead_server_wrong_process() {
+        let tmp_dir = tempfile::tempdir().unwrap();
+        let tmp_path = tmp_dir.path().to_owned();
+
+        let proc = tokio::process::Command::new(NODE_EXE)
+            .stdout(Stdio::null())
+            .stderr(Stdio::null())
+            .arg("-e")
+            .arg("setInterval(() => {}, 1000)")
+            .spawn()
+            .unwrap();
+
+        let pid = pid_path(&tmp_path);
+        std::fs::write(&pid, proc.id().unwrap().to_string()).unwrap();
+        let sock = sock_path(&tmp_path);
+        std::fs::write(&sock, "").unwrap();
+
+        let connector = DaemonConnector {
+            pid_file: pid,
+            sock_file: sock,
+            can_kill_server: true,
+            can_start_server: false,
+        };
+
+        let kill_pid = Pid::from(usize::MAX);
+        let proc_id = Pid::from(proc.id().unwrap() as usize);
+
+        assert_matches!(
+            connector.kill_dead_server(kill_pid).await,
+            Err(DaemonConnectorError::WrongPidProcess(daemon, running)) if daemon == kill_pid && running == proc_id
+        );
+    }
+
+    #[tokio::test]
+    async fn handles_kill_dead_server() {
+        let tmp_dir = tempfile::tempdir().unwrap();
+        let tmp_path = tmp_dir.path().to_owned();
+
+        let proc = tokio::process::Command::new(NODE_EXE)
+            .stdout(Stdio::null())
+            .stderr(Stdio::null())
+            .arg("-e")
+            .arg("setInterval(() => {}, 1000)")
+            .spawn()
+            .unwrap();
+
+        let pid = pid_path(&tmp_path);
+        std::fs::write(&pid, proc.id().unwrap().to_string()).unwrap();
+        let sock = sock_path(&tmp_path);
+        std::fs::write(&sock, "").unwrap();
+
+        let connector = DaemonConnector {
+            pid_file: pid,
+            sock_file: sock,
+            can_kill_server: true,
+            can_start_server: false,
+        };
+
+        assert_matches!(
+            connector
+                .kill_dead_server(Pid::from(proc.id().unwrap() as usize))
+                .await,
+            Ok(())
+        );
+    }
+}
```

Comment:
Is it possible to port `TestKillLiveServer` from Go as well?

Reply:
Yes, but I was planning on doing that when porting the server to rust because trying to launch a go server (and potentially rust binary) and all the path shenanigans around that feels like a nightmare given how short-lived it'll be.

Reply:
I will do it in this PR